bitacross-worker/bitacross/core/bc-task-receiver/src/lib.rs

```diff
@@ -50,44 +58,81 @@ pub enum Error {
 
 pub struct BitAcrossTaskContext<
 	ShieldingKeyRepository,
+	EthereumKeyRepository,
+	BitcoinKeyRepository,
 	S: StfEnclaveSigning<TrustedCallSigned>,
 	H: HandleState,
 	O: EnclaveOnChainOCallApi,
 > where
 	ShieldingKeyRepository: AccessKey,
+	EthereumKeyRepository: AccessKey<KeyType = EcdsaPair>,
+	BitcoinKeyRepository: AccessKey<KeyType = SchnorrPair>,
 	<ShieldingKeyRepository as AccessKey>::KeyType: ShieldingCryptoEncrypt + 'static,
 {
 	pub shielding_key: Arc<ShieldingKeyRepository>,
+	pub ethereum_key_repository: Arc<EthereumKeyRepository>,
+	pub bitcoin_key_repository: Arc<BitcoinKeyRepository>,
 	pub enclave_signer: Arc<S>,
 	pub state_handler: Arc<H>,
 	pub ocall_api: Arc<O>,
 }
 
 impl<
 		ShieldingKeyRepository,
+		EthereumKeyRepository,
+		BitcoinKeyRepository,
 		S: StfEnclaveSigning<TrustedCallSigned>,
 		H: HandleState,
 		O: EnclaveOnChainOCallApi,
-	> BitAcrossTaskContext<ShieldingKeyRepository, S, H, O>
+	> BitAcrossTaskContext<ShieldingKeyRepository, EthereumKeyRepository, BitcoinKeyRepository, S, H, O>
 where
 	ShieldingKeyRepository: AccessKey,
+	EthereumKeyRepository: AccessKey<KeyType = EcdsaPair>,
+	BitcoinKeyRepository: AccessKey<KeyType = SchnorrPair>,
 	<ShieldingKeyRepository as AccessKey>::KeyType: ShieldingCryptoEncrypt + 'static,
 	H::StateT: SgxExternalitiesTrait,
 {
 	pub fn new(
 		shielding_key: Arc<ShieldingKeyRepository>,
+		ethereum_key_repository: Arc<EthereumKeyRepository>,
+		bitcoin_key_repository: Arc<BitcoinKeyRepository>,
 		enclave_signer: Arc<S>,
 		state_handler: Arc<H>,
 		ocall_api: Arc<O>,
 	) -> Self {
-		Self { shielding_key, enclave_signer, state_handler, ocall_api }
+		Self {
+			shielding_key,
+			ethereum_key_repository,
+			bitcoin_key_repository,
+			enclave_signer,
+			state_handler,
+			ocall_api,
+		}
 	}
 }
 
-pub fn run_bit_across_handler_runner<ShieldingKeyRepository, S, H, O>(
-	_context: Arc<BitAcrossTaskContext<ShieldingKeyRepository, S, H, O>>,
+pub fn run_bit_across_handler_runner<
+	ShieldingKeyRepository,
+	EthereumKeyRepository,
+	BitcoinKeyRepository,
```

Comment:
Do you fancy using a shorter name to increase the readability?
Like SR, ER, BR ..

Reply:
Renamed 👍🏼